src/models.jl

```diff
@@ -0,0 +1,124 @@
+         dt (float) : time step
+ start_time (float) : start time for the simulation
+  time_step (float) : ?
+         arch (sym) : architecture (:cpu or :gpu)
+  float_type (type) : floating point type for model data (typically Float32 or Float64)
+          constants : planetary constants (?)
+                eos : equation of state to infer density from temperature and salinity
+            forcing : forcing functions for (u, v, w, T, S)
+boundary_conditions : boundary conditions
+     output_writers : output writer
+        diagonstics : diagnostics
+
+        ... and more.
+"""
+function Model(;
+    # Model resolution and domain size
+             N = (32, 1, 32),
+             L = (1.0, 1.0, 1.0),
+    # Molecular parameters
+             ν = 1.05e-6,
+            νh = ν,
+            νv = ν,
+             κ = 1.43e-7,
+            κh = κ,
+            κv = κ,
+    # Time stepping
+            dt = 1e-16,
+    start_time = 0,
+     time_step = 0, # ?
+    # Model architecture and floating point precision
+          arch = :cpu,
+    float_type = Float64,
+     constants = Earth(),
+    # Equation of State
+           eos = LinearEquationOfState(),
+    # Forcing and boundary conditions for (u, v, w, T, S)
+       forcing = Forcing(nothing, nothing, nothing, nothing, nothing),
+    boundary_conditions = BoundaryConditions(:periodic, :periodic, :rigid_lid, :free_slip),
+    # Output and diagonstics
+    output_writers = OutputWriter[],
+       diagnostics = Diagnostic[]
+)
+
+    # Initialize model basics.
+         metadata = ModelMetadata(arch, float_type)
+    configuration = ModelConfiguration(νh, νv, κh, κv)
+             grid = RegularCartesianGrid(metadata, N, L)
+            clock = Clock(start_time, time_step, dt)
+
+    # Initialize fields, including source terms and temporary variables.
+      velocities = VelocityFields(metadata, grid)
+         tracers = TracerFields(metadata, grid)
+       pressures = PressureFields(metadata, grid)
+               G = SourceTerms(metadata, grid)
+              Gp = SourceTerms(metadata, grid)
+        forcings = ForcingFields(metadata, grid)
+     stepper_tmp = StepperTemporaryFields(metadata, grid)
+    operator_tmp = OperatorTemporaryFields(metadata, grid)
+
+    # Initialize Poisson solver.
+    if metadata.arch == :cpu
+        stepper_tmp.fCC1.data .= rand(metadata.float_type, grid.Nx, grid.Ny, grid.Nz)
+        ssp = SpectralSolverParameters(grid, stepper_tmp.fCC1, FFTW.MEASURE; verbose=true)
+    elseif metadata.arch == :gpu
+        stepper_tmp.fCC1.data .= CuArray{Complex{Float64}}(rand(metadata.float_type, grid.Nx, grid.Ny, grid.Nz))
+        ssp = SpectralSolverParametersGPU(grid, stepper_tmp.fCC1)
+    end
+
+    # Default initial condition
+    velocities.u.data .= 0
+    velocities.v.data .= 0
+    velocities.w.data .= 0
+    tracers.S.data .= 35
+    tracers.T.data .= 283
+
+    # ?
+    pHY_profile = [-eos.ρ₀*constants.g*h for h in grid.zC]
+
+    # ?
```

Comment:
Don't like this part but yeah it just uses the hydrostatic pressure profile in `pHY_profile` to initialize `pressures.pHY`. We can probably get rid of storing `pHY` altogether as I don't think it's even used...

Reply:
I have no opinion as I do not know what the role of this variable is (hence the `?`).